index.js

```diff
@@ -8,17 +8,18 @@ const crypto = require('crypto')
 const Promise = require('bluebird')
 const sqs = require('sqs')
 
-const EVENTS = {
-  BOUNCE: 'bounce',
-  DELIVERED: 'delivered',
-  DROPPED: 'dropped',
-  SPAM: 'spamreport'
+const { AUTH, SQS_SUFFIX, PROVIDER, VALIDATION_KEY } = process.env
+
+if (! AUTH || ! SQS_SUFFIX || ! PROVIDER) {
+  throw new Error('Missing config')
 }
 
-const { AUTH, SQS_SUFFIX } = process.env
+if (PROVIDER !== 'sendgrid' && PROVIDER !== 'socketlabs') { 
```

Comment:
Let's make these provider strings constants, like `EVENTS` is in the other two modules.